Cllog/Projects/CllogService/Features/EditFeature/Sources/Presentation/ThumbnailsView.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  ThumbnailsView.swift
+//  EditFeature
+//
+//  Created by seunghwan Lee on 3/9/25.
+//  Copyright © 2025 Supershy. All rights reserved.
+//
+
+import SwiftUI
+import AVKit
+
+/// 편집 섬네일 컷 (10장)
+struct ThumbnailsView: View {
+    let asset: AVAsset
+    let duration: Double
+    let thumbnailSize: CGSize
+    let thumbnailCount: Int
+    
+    init(asset: AVAsset, duration: Double, frameSize: CGSize) {
+        self.thumbnailCount = 10
+        self.asset = asset
+        self.duration = duration
+        self.thumbnailSize = CGSize(width: frameSize.width / CGFloat(thumbnailCount), height: frameSize.height)
+    }
+    
+    @State private var thumbnails: [UIImage] = []
+    
+    public var body: some View {
+        HStack(spacing: 0) {
+            ForEach(thumbnails.indices, id: \.self) { index in
+                Image(uiImage: thumbnails[index])
+                    .resizable()
+                    .frame(width: thumbnailSize.width, height: thumbnailSize.height)
+                    .aspectRatio(contentMode: .fill)
+            }
+        }
+        .onAppear {
+            generateThumbnails()
+        }
+    }
+    
+    private func generateThumbnails() {
+        let generator = AVAssetImageGenerator(asset: asset)
+        generator.appliesPreferredTrackTransform = true
+        
+        Task { @MainActor in
+            let assetDuration = try await asset.load(.duration)
+            let durationSeconds = assetDuration.seconds
+            
+            let cmTimes = Array(0..<thumbnailCount).map { i in
+                let time = durationSeconds * Double(i) / Double(thumbnailCount)
+                return CMTime(seconds: time, preferredTimescale: 600)
+            }
+
+            var thumbnails: [UIImage] = []
+            for await imageResult in generator.images(for: cmTimes) {
+                if let image = try? imageResult.image {
+                    thumbnails.append(UIImage(cgImage: image))
+                }
+            }
+            
+            self.thumbnails = thumbnails
+        }
```

Comment:
```suggestion
        Task {
            let assetDuration = try await asset.load(.duration)
            let durationSeconds = assetDuration.seconds
            
            let cmTimes = Array(0..<thumbnailCount).map { i in
                let time = durationSeconds * Double(i) / Double(thumbnailCount)
                return CMTime(seconds: time, preferredTimescale: 600)
            }

            var thumbnails: [UIImage] = []
            for await imageResult in generator.images(for: cmTimes) {
                if let image = try? imageResult.image {
                    thumbnails.append(UIImage(cgImage: image))
                }
            }
            
            await MainActor.run {
                self.thumbnails = thumbnails
            }
        }
```
썸네일이 많은 경우에는 메인스레드가 낭비 될 것 같아서 UI 업데이트에만 메인 스레드 사용하면 좋을 것 같은데 이런 방법으로 하는건 어떻게 생각해?

Reply:
무조건 10장이긴 한데, 굳이 메인스레드에서 돌릴 필요는 없긴 하지 오키오키! 리뷰 감사